### ISSUE TITLE
Learning Rate is 0 returned by scheduler.get_last_lr()

### ISSUE THREAD

**Issue by wiekern** [luca-medeiros/lightning-sam#16]
We are using stepwise LR scheduler, however, the param step in **lr_lambda** function is 0 at the first time, so it will return **0** as a scalar multiplied by init_lr. If I were correct, whatever init_lr we have, the result seems to be 0 during the whole training process. At least in my own training process, the lr is awalys 0 by invoking API *scheduler.get_last_lr()*. 
```
  def lr_lambda(step):
      if step < cfg.opt.warmup_steps:
          return step / cfg.opt.warmup_steps
      elif step < cfg.opt.steps[0]:
          return 1.0
      elif step < cfg.opt.steps[1]:
          return 1 / cfg.opt.decay_factor
      else:
          return 1 / (cfg.opt.decay_factor**2)
```

The learning rate I extract from **optimizer.param_groups** varies stepwise and greater than 0. Is there any different in ways getting current learning rate between 
scheduler.get_last_lr() and optimizer.param_groups[0]['lr']? I read an [issue](https://github.com/Lightning-AI/lightning/issues/5363)  about this question, which tells me these are same up torch1.3

**Comment by luca-medeiros** [comment]
Interesting. Thanks for pointing it out.

I need to double-check the scheduler as I've used an existing code without checking much in-depth. Mind writing a PR?

**Comment by wiekern** [comment]
My pleasure, I am digging into the logic of  computing current learning rates,  hopefully I figure it out before you cope with it, then I will do some tests and  write a PR.

**Comment by wiekern** [comment]
[    model, optimizer, scheduler = fabric.setup(model, optimizer, scheduler)](https://github.com/luca-medeiros/lightning-sam/blob/80ac78866afa8ffc2f78ec2c314258b100589df0/lightning_sam/train.py#LL160C76-L160C76)

I am not sure if we should pass **scheduler** to fabric.setup as well, since the the type of scheduler is not **Optimizer**. 
I did two test cases as below,
1. model, optimizer = fabric.setup(model, optimizer), and *scheduler.get_last_lr()* gives me the varying value stepwise.
2. model, optimizer, scheduler = fabric.setup(model, optimizer, scheduler), *scheduler.get_last_lr()* gives me 0 all the time.

**Comment by luca-medeiros** [comment]
@wiekern 
Nice catch. It indeed shouldn't be passed to setup like that. (Confirmed by talking with a person from Lightning AI team)

Wanna PR a fix and maybe some changes to the scheduler?

**Comment by wiekern** [comment]
alright,  I have no clue whether we should do more changes on scheduler if we leave it outside *fabric.setup*, but we could first remove scheduler from *fabric.setup* to avoid the obvious error. 
One more thing, I notice that the FocalLoss is not consistent with the one at [focal_loss](https://github.com/facebookresearch/fvcore/blob/main/fvcore/nn/focal_loss.py), if you look into [code](https://github.com/luca-medeiros/lightning-sam/blob/80ac78866afa8ffc2f78ec2c314258b100589df0/lightning_sam/train.py#LL96C20-L96C20), you will find the balanced parameter alpha in focal loss is covered by the number of masks that is obviously not in range[0,1]. Could it be a typo?

**Comment by luca-medeiros** [comment]
@wiekern created another issue for it. Cheers
